STM32F1/libraries/SPI/src/SPI.cpp

```diff
@@ -385,6 +385,26 @@ uint16_t SPIClass::transfer16(uint16_t data) const
     return ret;
 }
 
+void SPIClass::transfer(void *buf, size_t count) 
+ {
+    if (count == 0) return;
+    spi_dev * spi_d = _currentSetting->spi_d;
+    uint8 *p = (uint8_t *)buf;
```

Comment:
This new pointer makes no sense, it is better to specify the buf parameter as (uint8_t *), as it can only transfer 8 bit data anyway.

Reply:
1)You're right it doesn't make any sense. It is like that because when i was trying to make my ADNS3080 work i was in a bit of a hurry and i basically copied the code from the spi.cpp of the original arduino. 
2)I thought you guys were gonna just alias the read() function. I tried the read() function and it works just fine(maybe better but my oscilloscope didn't show any difference). Plus this code is inefficient anyway and you guys pointed that out already so i m not really sure why you included my code into the main stm32 branch.